contentcuration/contentcuration/static/js/edit_channel/channel_set/views/ChannelSetList.vue

```diff
@@ -0,0 +1,266 @@
+      </div>
+    </div>
+    <div id="selectChannelsButton">
+      <button
+        class="action-button uppercase"
+        @click="goToSelectChannels"
+      >
+        <i class="material-icons">add</i>
+        {{ $tr('selectButtonLabel') }}
+      </button>
+    </div>
+
+  </div>
+
+</template>
+
+
+<script>
+
+import { mapGetters, mapActions } from 'vuex';
+import ChannelItem from './ChannelItem.vue';
+
+const copyStatusCodes = {
+  IDLE: "IDLE",
+  SUCCESS: "SUCCESS",
+  FAILED: "FAILED",
+}
+
+export default {
+  name: 'ChannelSetList',
+  $trs: {
+    'loading': "Loading...",
+    'titleLabel': "Title",
+    'titlePlaceholder': "Title your collection",
+    'descriptionLabel': "Description",
+    'descriptionPlaceholder': "Describe your collection",
+    'selectButtonLabel': 'Select',
+    'channelCountText': '{channelCount, plural, =1 {# channel in your collection} other {# channels in your collection}}',
+    'copyTokenButtonLabel': "Copy Token",
+    'titleRequiredText': "Title is required",
+    "charCount": "{charCount, plural, =1 {# character left} other {# characters left}}"
+  },
+  mounted() {
+    this.loadChannelSetChannels();
+  },
+  components: {
+    ChannelItem,
+  },
+  data() {
+    return {
+      copyStatus: copyStatusCodes.IDLE,
+    }
+  },
+  computed: Object.assign(
+    mapGetters('channel_set', [
+      'channels',
+      'channelSet',
+      'isValid',
+      'loadChannels'
+    ]),
+    {
+      name: {
+        get () {
+          return this.$store.state.channel_set.name;
+        },
+        set (value) {
+          this.$store.commit('channel_set/SET_NAME', value);
+        }
+      },
+      description: {
+        get () {
+          return this.$store.state.channel_set.description;
+        },
+        set (value) {
+          this.$store.commit('channel_set/SET_DESCRIPTION', value);
+        }
+      },
+      copyIcon() {
+        switch(this.copyStatus) {
+          case copyStatusCodes.SUCCESS:
+            return "check"
+          case copyStatusCodes.FAILED:
+            return "clear"
+          default:
+            return "content_paste"
+        }
+      },
+      charsLeft() {
+        return this.charLimit - this.description.length;
+      },
+      charLimit() {
+        return 400;
+      },
+      token() {
+        let token = this.channelSet.get('secret_token');
+        return token && token.display_token;
+      },
+      channelCount() {
+        return this.channels.length;
+      }
+    }
+  ),
+  methods: Object.assign(
+    mapActions('channel_set', [
+      'goToSelectChannels',
+      'loadChannelSetChannels'
+    ]),
+    {
+      copyToken() {
+        let element = this.$refs.tokenText;
+        element.select();
+        var self = this;
+        try {
+          document.execCommand("copy");
+          this.copyStatus = copyStatusCodes.SUCCESS;
+        } catch(e) {
+          this.copyStatus = copyStatusCodes.FAILED;
+        }
+        setTimeout(() => {
```

Comment:
What determines whether it is idle or not? Is this not driven by the backend, or is this just for UI persistence for a certain length of time?

Reply:
Mostly just for UI persistence- it isn't really used anywhere outside of the component